### PR TITLE
feat!: add KeyringExecutionContext to user ops methods

### PR DESCRIPTION
**Pull request by ccharly**
## Requires

- [x] https://github.com/MetaMask/keyring-api/pull/275

## Description

Adding an additional parameter about the "execution context" (contains only `chainId` for now) for User operations methods. This will also help computing the CAIP-2 scope/chain-id for underlying calls later on.

### Related to

- [ ] metamask/accounts-planning/issues/307

### CODE REVIEW

File: src/caip.ts [hunk 1]
```diff
@@ -16,6 +16,39 @@ export const CaipNamespaces = {
   Eip155: 'eip155' as CaipNamespace,
 } as const;
 
+/** Function signature for CAIP reference normalizers. */
+export type CaipReferenceNormalizer = (reference: string) => string;
+
+/**
+ * Normalize a eip155 CAIP reference.
+ *
+ * @param reference - The eip155 CAIP reference to be normalized.
```
Comment (danroc) [inline src/caip.ts:25]:
```suggestion
 * @param reference - The EIP-155 CAIP reference to be normalized.
```

File: src/caip.ts [hunk 2]
```diff
@@ -16,6 +16,39 @@ export const CaipNamespaces = {
   Eip155: 'eip155' as CaipNamespace,
 } as const;
 
+/** Function signature for CAIP reference normalizers. */
+export type CaipReferenceNormalizer = (reference: string) => string;
+
+/**
+ * Normalize a eip155 CAIP reference.
+ *
```
Comment (danroc) [inline src/caip.ts:24]: Can you add a detailed description + some examples to the JSDoc please?

File: src/SnapKeyring.test.ts [hunk 3]
```diff
@@ -59,6 +61,10 @@ describe('SnapKeyring', () => {
     },
   ] as const;
 
+  const keyringContext: KeyringExecutionContext = {
```
Comment (danroc) [inline src/SnapKeyring.test.ts:64]:
```suggestion
  const executionContext: KeyringExecutionContext = {
```

File: src/caip.ts (hunk 1 above)
Comment (ccharly) [reply src/caip.ts]: After discussing together, I removed this part!

File: src/SnapKeyring.test.ts (hunk 3 above)
Comment (ccharly) [reply src/SnapKeyring.test.ts]: Yup, it's better, I've changed everywhere, thanks!

File: src/caip.ts (hunk 2 above)
Comment (ccharly) [reply src/caip.ts]: After discussing together, I removed this part!

**Review by danroc — APPROVED** [review]

**Pull request merged**